### PR TITLE
Bugfix: Parameters were null after zip_longest

### PR DESCRIPTION
**Pull request by nagkumar91**
# Description

The SDK retrieves templates and parameters from the service. The SDK has some logic to merge all the template parameters via `zip_longest` to make sure that one template/parameter combo comes from each bucket.
`zip_longest` creates empty objects to match the lengths to the longest list.
Hence a few users saw some errors while running the adversarial simulator for 1000+ results.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Review by slister1001 — APPROVED** [review]

File: sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py [hunk 1]
```diff
@@ -221,6 +221,11 @@ async def __call__(
             random.shuffle(templates)
         parameter_lists = [t.template_parameters for t in templates]
         zipped_parameters = list(zip_longest(*parameter_lists))
+        filtered_parameters = []
```
Comment (diondrapeck) [inline sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py:224]: So we basically filter out all of the zipped pairs where one element of the pair is None (aka the shortest parameter list has run out)?

Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py:224]: yup, if we use `zip` the merge happens on the smallest list and we'll lose some parameters.

Comment (diondrapeck) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py:224]: Doesn't this code have the same effect as `zip` though? If we're making zipped_parameters equal to only the params that don't have a None in the pair (i.e. only ones before the second list runs out).

Let me know if I'm misunderstanding what this chunk does!

Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py:224]: if the lists being merged are of different lengths, `zip` would merge the lists taking the shortest length and exclude the items beyond that length.

Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py:224]: `list(zip([1, 2, 3], ['a', 'b']))  # Output: [(1, 'a'), (2, 'b')]`
```list(zip_longest([1, 2, 3], ['a', 'b']))  
# Output: [(1, 'a'), (2, 'b'), (3, None)]```

Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py:224]: But you might be right. I'll give it a try to use `zip` instead and check if that results in the same output.

Comment (nagkumar91) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py]: yup, you're right. `zip` does the job as expected.

Comment (diondrapeck) [reply sdk/evaluation/azure-ai-evaluation/azure/ai/evaluation/simulator/_adversarial_simulator.py]: Awesome!

**Review by diondrapeck — APPROVED** [review]

**Pull request merged**